### DISCUSSION TITLE
Option to list "Accept and Save" before suggested corrections

### DISCUSSION THREAD

**Discussion by ksafford** [minad/jinx#128] [Ideas]
When there are a large number of corrections, the options to Accept and Save the word in a personal dictionary, for the file, or for the session can be hidden. 

For new users, it may not be obvious that those options exist. And anyhow, a user might simply prefer to have those listed first. 

A custom variable like `jinx--suggestions-first` controlling the order these sections are pushed to the menu could be a nice option.

With a modest amount of guidance about how you'd like to handle this, I believe I could submit a branch to effect this.

**Comment by ksafford** [comment]
My initial thought was to create two functions; one to push the suggested words onto the menu, one to push the Accept and Save selections onto the menu and optionally call them in one order vs the other based on the flag. 

I _think_ this only needs to happen in `jinx--correct-menu`, but I haven't read every line of the code, so I could be wrong.

**Reply by minad** [reply]
Hi @ksafford!

Thank you for the suggestion. I am a bit hesitant to make this configurable, since it seems like a detail (hardly worth configuration) and I want to avoid adding unnecessary complexity.

**Reply by ksafford** [reply]
I understand. Allow me to make the case briefly. 

Personally, I've found myself using "Session" or "Personal" quite frequently. However, I use the spellchecker infrequently enough that I don't have the keys associated to those options memorized. So I first have to `M->` (end-of-buffer) to make those choices visible. In the screenshots in the README, it's not necessary because only a small number of options for correction are listed. But in practice, I'm getting dozens, so the Accept and Save block is always hidden from view.

**Reply by minad** [reply]
Pressing M-> is a small price to pay and imho doesn't justify making this configurable with an ad-hoc option. When I use Jinx I mostly select from one of the offered corrections, so moving the special options to the front would feel noisy. The issue with making such minute details configurable is that there is no clear line on where to stop. 

I think I would like to make this configurable if it would come as a side effect of a generalization of jinx--save-keys, which hopefully would not increase complexity. Right now jinx--save-keys only configures the special actions but one could imagine generalizing this somehow to a variable jinx--correct-menu, for example like this:

~~~ elisp
(defvar jinx--correct-menu
  `(dict-suggestions
    session-suggestions
    (save-personal ?@)
    (save-file ?*)
    (save-session ?+)))
~~~

**Reply by minad** [reply]
For more background - the jinx--save-keys variable is also private right now since the way to configure this menu is not yet fixed. Just recently I added mouse support which wasn't completely natural and somewhat conflicted with the existing design. If we add another variable jinx--suggestions-first, it will make it harder to change this again. If we instead go with a generalization as described before we end up with something cleaner and more flexible, maybe allowing for further tweaks by the user.

**Reply by ksafford** [reply]
Wanted to say I am liking this package. Best spellcheck UI I've found.

**Reply by minad** [reply]
Thanks!
